File: src/test/java/no/nav/sosialhjelp/soknad/business/service/oppsummering/steg/OppsummeringTestUtils.java
package no.nav.sosialhjelp.soknad.business.service.oppsummering.steg;

import no.nav.sosialhjelp.soknad.web.rest.ressurser.oppsummering.dto.Felt;
import no.nav.sosialhjelp.soknad.web.rest.ressurser.oppsummering.dto.SvarType;
import no.nav.sosialhjelp.soknad.web.rest.ressurser.oppsummering.dto.Type;

import static org.assertj.core.api.Assertions.assertThat;

public final class OppsummeringTestUtils {

    public static void validateFeltMedSvar(Felt felt, Type type, SvarType svarType, String svarValue) {
        assertThat(felt.getType()).isEqualTo(type);
        assertThat(felt.getSvar().getType()).isEqualTo(svarType);
        assertThat(felt.getSvar().getValue()).isEqualTo(svarValue);
    }

    private OppsummeringTestUtils() {
        //no-op
    }
}

File: src/test/kotlin/no/nav/sosialhjelp/soknad/oppsummering/steg/OppsummeringTestUtils.kt
package no.nav.sosialhjelp.soknad.oppsummering.steg

import no.nav.sosialhjelp.soknad.oppsummering.dto.Felt
import no.nav.sosialhjelp.soknad.oppsummering.dto.SvarType
import no.nav.sosialhjelp.soknad.oppsummering.dto.Type
import org.assertj.core.api.Assertions.assertThat

internal object OppsummeringTestUtils {
    fun validateFeltMedSvar(felt: Felt, type: Type?, svarType: SvarType?, svarValue: String?) {
        assertThat(felt.type).isEqualTo(type)
        assertThat(felt.svar?.type).isEqualTo(svarType)
        assertThat(felt.svar?.value).isEqualTo(svarValue)
    }
}
